app/views/my_facilities/blood_pressure_control.html.erb

```diff
@@ -1,209 +1,318 @@
-                  data-year="<%= month.year %>"
-                  form="query-filters"
-                >
-                  <%= month_short_name_and_year(month) %>
-                </button>
+        </div>
+        <% %w(large medium small community).each do |size| %>
+          <% next if @facilities_by_size[size].blank? %>
+          <p class="mt-20px mb-0px fw-bold">
+            <%= size.capitalize %> facilities
+          </p>
+          <table class="mt-3 mt-lg-4 table table-compact table-responsive-md table-hover analytics-table">
+            <colgroup>
+              <col>
+              <col>
+              <col>
+              <col class="table-divider">
+              <col>
+              <col class="table-divider">
+              <col>
+              <col class="table-divider">
+              <col>
+              <col class="table-divider">
+              <col>
+              <col class="table-divider">
+              <col>
+              <col class="table-divider">
+              <col>
+              <col class="table-divider">
+              <col>
+            </colgroup>
+            <thead>
+              <tr data-sort-method="thead" class="sorts">
+                <th class="row-label sort-label sort-label-small" style="width: 160px;">
+                  Facilities
+                </th>
+                <th class="row-label sort-label sort-label-small" data-sort-method="number" style="width: 120px;">
+                  Total assigned<br>patients
+                </th>
+                <th class="row-label sort-label sort-label-small" data-sort-method="number" style="width: 120px;">
+                  Total registered<br>patients
+                </th>
+                <th class="row-label sort-label sort-label-small" colspan="2" style="width: 80px;">
+                  6-month change
+                </th>
+                <% (@start_period..@period).each do |period| %>
+                  <th class="row-label sort-label sort-label-small" data-sort-method="number" colspan="2">
+                    <%= period %>
+                  </th>
+                <% end %>
+              </tr>
+            </thead>
+            <tbody>
+              <% @facilities_by_size[size].each do |facility| %>
+                <% six_month_rate_change = @data_for_facility[facility.name]["controlled_patients_rate"][@period] - @data_for_facility[facility.name]["controlled_patients_rate"][@start_period] || 0 %>
+                <tr data-row="<%= facility.slug %>" data-trend-color="<%= six_month_rate_change >= 0 ? "green" : "red" %>">
+                  <td class="type-title">
+                    <%= link_to(reports_region_path(facility, report_scope: "facility")) do %>
+                      <%= facility.name %>
+                    <% end %>
+                  </td>
+                  <td>
+                    <%= number_with_delimiter(@data_for_facility[facility.name]["adjusted_registrations"].values.last || 0) %>
```

Comment:
We should add a new helper function for this - it can follow the pattern set here: 

https://github.com/simpledotorg/simple-server/blob/85b004c202e0b79353896f2462eea75fa8c52391/app/helpers/dashboard_helper.rb#L2-L5

Reply:
And any other `number || 0` can use the same helper so we don't have to have this logic in the view